tests/src/Integration/Action/SystemPageRedirectTest.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\rules\Integration\Action\SystemSendEmailTest.
+ */
+
+namespace Drupal\Tests\rules\Integration\Action;
+
+use Drupal\Tests\rules\Integration\RulesIntegrationTestBase;
+
+/**
+ * @coversDefaultClass \Drupal\rules\Plugin\Action\SystemPageRedirect
+ * @group rules_actions
+ *
+ * @todo: Write test for redirection to rules admin page
+ * @todo: Write test for redirection to external link
+ * @todo: Write test for redirection with destination
+ */
+class SystemPageRedirectTest extends RulesIntegrationTestBase {
+
+  /**
+   * @var \Psr\Log\LoggerInterface
+   */
+  protected $logger;
+
+  /**
+   * @var \Drupal\Core\Routing\RedirectDestinationInterface
+   */
+  protected $redirectDestination;
+
+  /**
+   * The action to be tested.
+   *
+   * @var \Drupal\rules\Plugin\Action\SystemPageRedirect
+   */
+  protected $action;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+
+    require_once __DIR__ . '/../../../../../../core/includes/form.inc';
```

Comment:
Aha. I understand your problem now. Find me in the rules sprint room so I can show you how to do this.